Update cosmiconfig module to latest upstream release

With cosmiconfig 7.0.1, build fails with the following error. I'm packaging webfont for debian and would like to use cosmiconfig 7.0.1 from debian archive (we prefer to use only one version of a module - in rare cases, we can use multiple versions but we prefer to avoid such cases if possible).

```
$ npm run test

> webfont@11.3.0 pretest
> npm run build


> webfont@11.3.0 prebuild
> npm run clean && npm run lint


> webfont@11.3.0 clean
> rm -rf dist/ temp/


> webfont@11.3.0 lint
> eslint .

Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating

> webfont@11.3.0 build
> rollup -c


src/cli/index.ts → dist/cli.js...
(!) Plugin typescript: @rollup/plugin-typescript TS2349: This expression is not callable.
  Type 'typeof import("/home/pravi/forge/js-team/upstream/webfont/node_modules/cosmiconfig/dist/index")' has no call signatures.
src/standalone/index.ts: (28:26)

28   const configExplorer = cosmiconfig("webfont");
                            ~~~~~~~~~~~

created dist/cli.js in 6.4s

src/index.ts → dist...
created dist in 153ms

> webfont@11.3.0 postbuild
> tsc --declaration --declarationDir dist --emitDeclarationOnly

src/standalone/index.ts:28:26 - error TS2349: This expression is not callable.
  Type 'typeof import("/home/pravi/forge/js-team/upstream/webfont/node_modules/cosmiconfig/dist/index")' has no call signatures.

28   const configExplorer = cosmiconfig("webfont");
                            ~~~~~~~~~~~


Found 1 error.


```

--- a/package.json
+++ b/package.json
@@ -37,7 +37,7 @@
     "testu": "npm test -- -u"
   },
   "dependencies": {
-    "cosmiconfig": "^5.2.0",
+    "cosmiconfig": "7.1.0",
     "deepmerge": "^4.2.2",
     "globby": "^11.0.0",
     "meow": "^9.0.0",

--- a/src/standalone/index.ts
+++ b/src/standalone/index.ts
@@ -1,4 +1,4 @@
-import cosmiconfig from "cosmiconfig";
+import { cosmiconfig } from "cosmiconfig";
 import crypto from "crypto";
 import deepmerge from "deepmerge";
 import globby from "globby";
@@ -18,28 +18,26 @@ import type { ResultConfig } from "../types/ResultConfig";
 import { getGlyphsData } from "./glyphsData";
 import { getOptions } from "./options";
 
-const buildConfig = async (options) => {
-  let searchPath = process.cwd();
-  let configPath = null;
+type CosmiconfigLoaded = NonNullable<Awaited<ReturnType<ReturnType<typeof cosmiconfig>["search"]>>>;
 
-  if (options.configFile) {
-    searchPath = null;
-    configPath = path.resolve(process.cwd(), options.configFile);
-  }
+const isCosmiconfigLoaded = (value: CosmiconfigLoaded | Record<string, never>): value is CosmiconfigLoaded =>
+  "filepath" in value;
 
+const buildConfig = async (options: {
+  configFile?: string;
+}): Promise<CosmiconfigLoaded | Record<string, never>> => {
   const configExplorer = cosmiconfig("webfont");
 
-  let config = await configExplorer.search(searchPath);
+  if (options.configFile) {
+    const configPath = path.resolve(process.cwd(), options.configFile);
+    const config = await configExplorer.load(configPath);
 
-  if (configPath) {
-    config = await configExplorer.load(configPath);
+    return config ?? {};
   }
 
-  if (!config) {
-    return {};
-  }
+  const config = await configExplorer.search(process.cwd());
 
-  return config;
+  return config ?? {};
 };
 
 const toSvg = (glyphsData, options) => {
@@ -96,7 +94,7 @@ export const webfont: Webfont = async (initialOptions) => {
 
   let discoveredConfigPath: string | undefined;
 
-  if (Object.keys(config).length > 0) {
+  if (isCosmiconfigLoaded(config)) {
     options = deepmerge(options, config.config, {
       arrayMerge: (_destinationArray, sourceArray) => sourceArray,
     });

